Apps/Examples/Examples/All Examples/Annotations/AnimatedMarkerExample.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+import MapboxMaps
+import UIKit
+
+@objc(AnimatedMarkerExample)
+final class AnimatedMarkerExample: UIViewController, ExampleProtocol {
+    enum Constants {
+        static let markerIconId = "marker_icon"
+        static let sourceId = "source-id"
+        static let animationDuration: CFTimeInterval = 2
+    }
+    private var mapView: MapView!
+
+    private var currentPosition = CLLocationCoordinate2D(latitude: 64.900932, longitude: -18.167040)
+
+    private var animationStartTimestamp: CFTimeInterval = 0
+    private var origin: CLLocationCoordinate2D!
+    private var destination: CLLocationCoordinate2D!
+    private var displayLink: CADisplayLink?
+
+    deinit {
+        displayLink?.invalidate()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let cameraOptions = CameraOptions(center: currentPosition, zoom: 5)
+        let initOptions = MapInitOptions(cameraOptions: cameraOptions)
+        mapView = MapView(frame: view.bounds, mapInitOptions: initOptions)
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        view.addSubview(mapView)
+
+        // Allows the delegate to receive information about map events.
+        mapView.mapboxMap.onNext(.mapLoaded) { [weak self] _ in
+
+            // Set up the example
+            self?.setupExample()
+
+            // The below line is used for internal testing purposes only.
+            self?.finish()
```

Comment:
should we let the animation run before finishing?

Reply:
This example doesn't animate anything on its own, do you suggest to animate marker position after the map is loaded?

Reply:
Ah, I see what you mean. No, it's fine as-is